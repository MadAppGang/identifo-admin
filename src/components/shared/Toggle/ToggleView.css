.iap-default-toggle {
  display: flex;
}

.iap-default-toggle > button {
  margin-right: 10px;
}

.iap-default-toggle > span {
  font: normal 15px 'Circe';
  color: #aaa;
}

.iap-default-toggle__body {
  height: 20px;
  width: 45px;
  border-radius: 20px;
  background-color: #eee;
  position: relative;
  display: flex;
  align-items: center;
  transition: all 200ms;
  cursor: pointer;
  border: 1px solid #eee;
  outline: none;
  box-sizing: border-box;
}

.iap-default-toggle__body.iap-default-toggle__body--on {
  background-color: #D8EEFF;
}

.iap-default-toggle__body:hover .iap-default-toggle__handle {
  left: 1px;
}

.iap-default-toggle__body.iap-default-toggle__body--on:hover .iap-default-toggle__handle {
  left: 100%;
  transform: translateX(calc(-100% - 1px));
}

.iap-default-toggle__handle {
  height: 27px;
  width: 27px;
  background-color: #fff;
  border: 1.5px solid #eee;
  left: 0;
  transform: translateX(0);
  position: absolute;
  border-radius: 50%;
  transition: all 200ms;
}

.iap-default-toggle__body.iap-default-toggle__body--on .iap-default-toggle__handle {
  background-color: #3CA6F5;
  border: 1.5px solid #D8EEFF;
  left: 100%;
  transform: translateX(-100%);
}