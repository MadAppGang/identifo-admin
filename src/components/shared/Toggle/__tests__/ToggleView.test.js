import ToggleView from '../View';

describe('<ToggleView />', () => {
  test('matches snapshot when turned off', () => {
    const props = {
      on: false,
      toggle: jest.fn(),
      label: 'Label',
    };

    expect(shallow(<ToggleView {...props} />)).toMatchSnapshot();
  });

  test('matches snapshot when turned on', () => {
    const props = {
      on: true,
      toggle: jest.fn(),
      label: 'Label',
    };

    expect(shallow(<ToggleView {...props} />)).toMatchSnapshot();
  })
});
