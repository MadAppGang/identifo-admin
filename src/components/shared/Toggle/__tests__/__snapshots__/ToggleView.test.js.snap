// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleView /> matches snapshot when turned off 1`] = `
<div
  className="iap-default-toggle"
>
  <button
    className="iap-default-toggle__body"
    onClick={[MockFunction]}
    type="button"
  >
    <div
      className="iap-default-toggle__handle"
    />
  </button>
  <span>
    Label
  </span>
</div>
`;

exports[`<ToggleView /> matches snapshot when turned on 1`] = `
<div
  className="iap-default-toggle"
>
  <button
    className="iap-default-toggle__body iap-default-toggle__body--on"
    onClick={[MockFunction]}
    type="button"
  >
    <div
      className="iap-default-toggle__handle"
    />
  </button>
  <span>
    Label
  </span>
</div>
`;
